### PR TITLE
Remove ref to E2E_MNEMONIC

### PR DESCRIPTION
**Pull request by vilsbole**
- Remove ref to E2E_MNEMONIC
- Remove dead code
- Upgrade testcafe and adapt to new api.

### CODE REVIEW

**Review by FrederikBolding — CHANGES_REQUESTED** [review]

File: .env.example [hunk 1]
```diff
@@ -1 +1 @@
-E2E_MNEMONIC_PASSPHRASE="mnemonic phrase here"
\ No newline at end of file
+E2E_PRIVATE_KEY="mnemonic phrase here"
```
[the remark this answers is not in the document]
Comment (vilsbole) [reply .env.example:1]: 👍

File: __tests__/ethereum-mock.js [hunk 2]
```diff
@@ -11,7 +11,12 @@ export const _setupEthereumMock = ClientFunction((privateKey, chainId, rpcUrl) =
 });
 
 export const setupEthereumMock = async (privateKey, chainId) => {
-  await t.expect(ClientFunction(() => window.initialize)).ok({ timeout: FIXTURES_CONST.TIMEOUT });
+  await t
+    // Make sure to call the client function in order to assert against its return value:
+    // e.g "You passed a ClientFunction object to 't.expect()'.
+    // If you want to check the function's return value, use parentheses to call the function: fnName()."
+    .expect(ClientFunction(() => window.ethereum)())
```
Comment (FrederikBolding) [inline __tests__/ethereum-mock.js:18]: We want to check that `initialize` is available here, as `window.ethereum` has not necessarily been injected yet.

**Review by FrederikBolding — CHANGES_REQUESTED** [review]

File: __tests__/ethereum-mock.js (hunk 2 above)
Comment (vilsbole) [reply __tests__/ethereum-mock.js]: `ClientFunction` return values are tricky.
Can't pass a reference to a window object to the caller, and cloning functionality inside the ClientFn are absent (Object.assign, JSON.stringify etc). 

I found a solution by passing a predicate to `waitForProperty`.

File: __tests__/ethereum-mock.js [hunk 3]
```diff
@@ -1,21 +1,20 @@
...
 });
 
 export const setupEthereumMock = async (privateKey, chainId) => {
   await t
-    // Make sure to call the client function in order to assert against its return value:
-    // e.g "You passed a ClientFunction object to 't.expect()'.
-    // If you want to check the function's return value, use parentheses to call the function: fnName()."
-    .expect(ClientFunction(() => window.ethereum)())
+    .expect(waitForProperty((w) => w.ethereum.initialize))
+    // .expect(ClientFunction(() => window.ethereum.initialize)())
```
Comment (FrederikBolding) [inline __tests__/ethereum-mock.js:17]: Delete unused code please 👍

**Review by FrederikBolding — CHANGES_REQUESTED** [review]

**Review by FrederikBolding — APPROVED** [review]
lgtm - good to merge when tests pass

**Pull request merged**